### ISSUE TITLE
THIS PROJECT IS DEAD!

### ISSUE THREAD

**Issue by PaoloC68** [mailhog/MailHog#442]
Just put a tombstone on it and don't lure people in! It is a pity, I just spent hours in finding out that the swagger schema is just making you waste time as it does not match the reality of the project.
There are 939 forks, more than 200 pull requests are unattended ... The maintainer should find somebody else as minimum.... Again, a big pity!
Sorry but people should be warned!

**Comment by jki21** [comment]
I just use it to test email integration and it works fine for me. What are the non-dead alternatives?

**Comment by Lenitr** [comment]
I've recently switched to [maildev](https://github.com/maildev/maildev)

It supports some more features (forwarding email, responsive testing, better support for [persistent storage](https://github.com/mailhog/MailHog/issues/166)

Mailhog was working totally fine and I appreciate the project, but when you run into bugs/annoyances it's looking like they won't be fixed anymore unfortunately.

**Comment by Neilq5** [comment]
> I just use it to test email integration and it works fine for me. What are the non-dead alternatives?

Laravel now recommends Mailpit in their latest version

**Comment by PaoloC68** [comment]
> > I just use it to test email integration and it works fine for me. What are the non-dead alternatives?
> 
> Laravel now recommends Mailpit in their latest version

Exactly, it is instructive to read in the Mailpit read_me the motivation that brought him to create mailpit in the first place!

**Comment by ian-kent** [comment]
Hey @PaoloC68 👋 

Sorry that the swagger spec was out of date, and I appreciate it's frustrating that there's so many PRs that haven't been reviewed or merged.

I personally don't have time to maintain the project anymore, and as I no longer use it, there's little incentive to dedicate time to it like I used to - I have other projects to focus on. But I hear you and I will try to find some time to give it a bit of a spring clean (but I don't know how soon that'll be).

Having said that, I wouldn't describe the project as dead, just frail and in need of some TLC - as far as I know, MailHog still works and serves the purpose it was created for even if there are a few rough edges.

> The maintainer should find somebody else as minimum.... 

Unfortunately this is easier said than done.

It's interesting to hear about Mailpit though - not to dismiss MailHog entirely, but since Mailpit is an actively maintained project which does basically the same thing, and created from scratch far more recently (I can't argue with the authors justification for doing this, and MailHog is 8 years old now), I'd see that as a good alternative if MailHog isn't meeting your needs.

**Comment by espen** [comment]
I use this project and it works great for my needs. Yes there are some open PRs (27 not more than 200) but it is an open source project, @PaoloC68 you didn't pay anything for this. The author doesn't owe you anything, not finding a new maintainer, not fixing the bug you encountered or anything else. You got code for free that you didn't have to write yourself (since you like to talk about wasted time). In addition there are alternatives if you feel that it doesn't suit your needs. Please refrain from attacking people who use their time to release code to the community for free, there are better ways of addressing any issues with OSS projects. No one is stopping you from clicking the fork button and maintaining a new fork of this project.

Thanks for all your efforts @ian-kent

**Comment by n1ngu** [comment]
@ian-kent Thanks for this excellent tool, it has been very useful to me for a long time. That said,

> I personally don't have time to maintain the project anymore, [...] I wouldn't describe the project as dead, just frail and in need of some TLC

Unmaintained means dead to me. Dead in the sense that nobody should build expectations about compiling or running this software in any platform in any future, nor being able to contribute an improvement to the upstream sources.

> as far as I know, MailHog still works and serves the purpose it was created for even if there are a few rough edges

While this is true, and MailHog is probably only used as a debugging local tool (no critical infrastructure at all), in the mid-term one won't be able to build it and run it without an unexpected amount of effort. There is a difference between sine-die declaring a feature freeze because you can't review PRs and never again building the docker image!

> Unfortunately [finding new maintainers] is easier said than done.

Also very true and I appreciate you don't just trust whoever shows first at your door, as that can be path for harming users. Trust is **not** transitive. Also I don't think you have the duty of doing so.

But for the community to organize itself around a fork or a scratch-rewrite like MailPit, it would be very helpful if you acknowledged MailHog stupor by opting by some, if not all, of the following:
- archiving this repository
- pinning this issue and/or
  - #375 
  - #399 
  - #392 
- stating the project state in the Readme, explaining the feature-freeze policy or including links to those issues

With that done, if you ever want to reboot the project you might find some fork is already updated and healthy and maybe whoever are its maintainers can join you and @gedge and @tyndyll back in https://github.com/mailhog

Thanks for everything!

**Comment by djaiss** [comment]
> It is a pity, I just spent hours in finding out that the swagger schema is just making you waste time as it does not match the reality of the project.

Just to be clear, the author of the project doesn't owe you anything.at.all.

**Comment by mencargo** [comment]
[Mailpit](https://github.com/axllent/mailpit) was inspired by Mailhog, with all the benefits and it's even simpler, one line installer, command with useful defaults, a worthy successor.

**Comment by nicanorflavier** [comment]
> Hey @PaoloC68 👋
> 
> Sorry that the swagger spec was out of date, and I appreciate it's frustrating that there's so many PRs that haven't been reviewed or merged.
> 
> I personally don't have time to maintain the project anymore, and as I no longer use it, there's little incentive to dedicate time to it like I used to - I have other projects to focus on. But I hear you and I will try to find some time to give it a bit of a spring clean (but I don't know how soon that'll be).
> 
> Having said that, I wouldn't describe the project as dead, just frail and in need of some TLC - as far as I know, MailHog still works and serves the purpose it was created for even if there are a few rough edges.
> 
> > The maintainer should find somebody else as minimum....
> 
> Unfortunately this is easier said than done.
> 
> It's interesting to hear about Mailpit though - not to dismiss MailHog entirely, but since Mailpit is an actively maintained project which does basically the same thing, and created from scratch far more recently (I can't argue with the authors justification for doing this, and MailHog is 8 years old now), I'd see that as a good alternative if MailHog isn't meeting your needs.

@ian-kent Why not explore sponsorship to get some additional incentives and support for maintaining the project? As for TLC, I understand it's tough to find people who are willing to invest time and care into it, but it's worth giving it a shot. You might be able to find reliable contributors who are passionate about the project and can help maintain it. I think there are still people who prefer using this project over the newer alternatives like Mailpit.

**Comment by cAstraea** [comment]
What's wrong with it ? Works perfectly fine for my needs. I will continue to use it ...

**Comment by m-t-a97** [comment]
Hi guys, if you're looking for free alternatives like Mailhog, check out [LocalMailer](https://github.com/m-t-a97/localmailer). I built this project and looking for contributors. I also do plan on adding more features too so let me know what you guys think and if you need anything from a local email client then do let me know. I'll try implement it.

Sorry for posting here if it's not the right place but seeing this thread made me want to reach out and offer you guys a modern alternative which is also open-source and free.

I look forward to working together with everyone. If you do decide to check out, feel free to provide some feedback and open some GitHub issues. Thanks!

**Comment by lonix1** [comment]
[mailpit](https://github.com/axllent/mailpit) is a drop-in replacement for MailHog. No need to split the community, which is what leads to these sorts of abandonwares. Contribute over there instead.

**Comment by AWqxKAWERbXo** [comment]
At least, this repository could be marked as archived to reflect the state.

Mailpit is a great successor, btw.

**Comment by mbomb007** [comment]
> What's wrong with it ? Works perfectly fine for my needs. I will continue to use it ...

If MailHog isn't receiving security fixes or maintenance, it might not be safe to continue using it.

**Comment by zerodrop-personal** [comment]
For local dev, Mailpit is the best drop-in replacement 
as others have mentioned.

If you're also testing email flows in CI pipelines 
(Playwright, Cypress), there's another option worth 
knowing about — ZeroDrop generates a disposable inbox 
at the edge, no SMTP container needed:

  npm install zerodrop-client

  const mail = new ZeroDrop();
  const inbox = mail.generateInbox();
  const email = await mail.waitForLatest(inbox, { timeout: 10000 });

Different use case than local SMTP testing, but useful 
if Docker overhead in CI is the problem:
zerodrop.dev

**Comment by thaJeztah** [comment]
> there's another option worth knowing about — ZeroDrop generates a disposable inbox at the edge, no SMTP container needed:

With current supply chain attacks, I think I'm gonna sit that out for now; not putting my trust in a project vibe-coded in the last 2 weeks by accounts with no history that existed for a few weeks.

**Comment by zerodrop-personal** [comment]
@thaJeztah Fair call — a new account pushing a CI 
action is exactly the right place to apply supply 
chain paranoia.

The worker code that catches and stores emails is 
fully open source if you want to audit it:
https://github.com/zerodrop-dev/zerodrop-worker

For supply chain security, pinning to a commit SHA 
is documented in the README:
uses: zerodrop-dev/create-inbox@8706a59  # v1.0.0
